### PR TITLE
Rem 625

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
new page : iv-user-defined
post and get DE values of ivtUDTCount , ivtUDT[index] , ivtDimCount , ivtDimension[index]

### CODE REVIEW

File: src/components/Inputs/CustomNumberField.js [hunk 1]
```diff
@@ -16,14 +16,17 @@ const CustomNumberField = ({
...
+  unClearable = false,
   error,
   helperText,
   hasBorder = true,
   maxLength = 1000,
   thousandSeparator = ',',
-  min = '',
-  max = '',
+  min,
+  max,
```
Comment (RebeccaFrem02) [inline src/components/Inputs/CustomNumberField.js:26]: why did you remove the =''? it was here for the case where we dont put min and max

File: src/components/Inputs/CustomNumberField.js [hunk 2]
```diff
@@ -84,11 +87,15 @@ const CustomNumberField = ({
       onInput={handleInput}
       InputProps={{
         inputProps: {
-          tabIndex: readOnly ? -1 : 0 // Prevent focus on the input field
+          min: min,
+          max: max,
+          step: step,
```
Comment (RebeccaFrem02) [inline src/components/Inputs/CustomNumberField.js:92]: what does step stand for here?

File: src/pages/iv-user-defined/forms/UserTextForm.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+      process.nextTick(() => {
+        clearExcessFields(stagingDimCount)
+      })
+    }
+  }
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} width={'50%'} sx={{ marginLeft: '0.5rem' }}>
```
Comment (RebeccaFrem02) [inline src/pages/iv-user-defined/forms/UserTextForm.js:145]: we need to work on this styling

File: src/pages/iv-user-defined/forms/UserDifinedForm.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  }
+
+  const { labels: _labels } = useResourceQuery({
+    datasetId: ResourceIds.UserDefined
+  })
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} width={'50%'} sx={{ marginLeft: '0.5rem' }}>
```
Comment (RebeccaFrem02) [inline src/pages/iv-user-defined/forms/UserDifinedForm.js:145]: need to work on the styling here

**Comment by RebeccaFrem02** [comment]
it came to my attention that both tabs are similar, wouldn't be better it we made one form and send the differences as props for each tab instead of rewriting the same code?

File: src/components/Inputs/CustomNumberField.js (hunk 1 above)
Comment (Alinadimmokdad) [reply src/components/Inputs/CustomNumberField.js:26]: we can gave it directly from the component  example : min={1}

File: src/components/Inputs/CustomNumberField.js (hunk 2 above)
Comment (Alinadimmokdad) [reply src/components/Inputs/CustomNumberField.js:92]: when  you decrease the value it decrease by step = 1

**Comment by omarZrayka23** [comment]
The label was not defined on the load page for the first time.

File: src/pages/iv-user-defined/forms/UserDefinedForm.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+const UserDefinedForm = ({ labels }) => {
+  const { postRequest } = useContext(RequestsContext)
+  const { platformLabels, defaultsData, updateDefaults } = useContext(ControlContext)
+  const [errored, setErrored] = useState(false)
```
Comment (omarZrayka23) [inline src/pages/iv-user-defined/forms/UserDefinedForm.js:19]: u can remove state and use direct  const errored=  !!formik.value.ivtDimCount

File: src/pages/iv-user-defined/forms/UserTextForm.js [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  const handleSubmit = () => {
+    if (errored) {
+      return
+    }
+    formik.handleSubmit()
+  }
+
+  const [stagingDimCount, setStagingDimCount] = useState(formik.values.ivtUDTCount || null)
```
Comment (omarZrayka23) [inline src/pages/iv-user-defined/forms/UserTextForm.js:104]: Could you please delete the state set value direct as a comment before?

File: src/pages/iv-user-defined/forms/UserTextForm.js [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+                    inputProps={{
+                      readOnly: errored || formik.values.ivtUDTCount <= index || formik.values.ivtUDTCount === 'null'
+                    }}
+                  />
+                </Grid>
+              ))}
+            </Grid>
+          </Grid>
+
+          <Grid item xs={12} md={6}>
```
Comment (omarZrayka23) [inline src/pages/iv-user-defined/forms/UserTextForm.js:178]: Why did you utilize two CustomTextFields? Only CustomTextField with loop can be used.

File: src/pages/iv-user-defined/forms/UserDefinedForm.js [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        ])
+      )
+    }),
+
+    onSubmit: async values => {
+      await postDimensionSettings(values)
+    }
+  })
+
+  // const [stagingDimCount, setStagingDimCount] = useState(formik.values.ivtDimCount || null)
```
Comment (omarZrayka23) [inline src/pages/iv-user-defined/forms/UserDefinedForm.js:53]: remove unused code

**Review by omarZrayka23 — APPROVED** [review]
